Website/src/rendering/src/components/Forms/CreditCardForm.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormEvent, useState } from 'react';
+import { getMockExpirationDate } from '../../stories/utils';
+import {
+  getIsoDateFromYearAndMonth,
+  getMonthFromIsoDateString,
+  getYearFromIsoDateString,
+} from '../../helpers/DateHelper';
+import { DBuyerCreditCard } from '../../models/ordercloud/DCreditCard';
+
+type CreditCardFormProps = {
+  creditCard?: DBuyerCreditCard;
+  onSubmit?: (payment: DBuyerCreditCard) => void;
+};
+
+const CreditCardForm = (props: CreditCardFormProps): JSX.Element => {
+  const [cardholderName, setCardholderName] = useState(
+    props?.creditCard?.ID ? props?.creditCard?.CardholderName || '' : 'John Smith' // TODO: remove mocked data once we have saved credit cards
+  );
+  const [cardNumber, setCardNumber] = useState(props?.creditCard?.ID ? '' : '4111111111111111'); // TODO: remove mocked data once we have saved credit cards
+  const [expirationMonth, setExpirationMonth] = useState(
+    getMonthFromIsoDateString(
+      props?.creditCard?.ID ? props?.creditCard?.ExpirationDate : getMockExpirationDate() // TODO: remove mocked data once we have saved credit cards
+    )
+  );
+  const [expirationYear, setExpirationYear] = useState(
+    getYearFromIsoDateString(
+      props?.creditCard?.ID ? props?.creditCard?.ExpirationDate : getMockExpirationDate() // TODO: remove mocked data once we have saved credit cards
+    )
+  );
+  const yearNow = new Date().getFullYear();
+  const expirationYearRange = new Array(11).fill('').map((_, index) => yearNow + index);
+  const expirationMonthRange = new Array(12)
+    .fill('')
+    .map((_, index) => `${index + 1}`.padStart(2, '0'));
+
+  const handleFormSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const updatedCreditCard: DBuyerCreditCard = {
+      ...(props.creditCard || {}),
+      CardType: 'Visa', // hardcoding for now
+      CardholderName: cardholderName,
+      PartialAccountNumber: cardNumber.slice(-4),
+      ExpirationDate: getIsoDateFromYearAndMonth(expirationYear, expirationMonth),
+      // in a real application we would use a third party payment processor
+      // to securely capture, store, and tokenize sensitive credit card details and store only the opaque token
+      // it is against the terms of service to store sensitive credit card details on the platform
+      Token: '',
+    };
+
+    if (props.onSubmit && typeof props.onSubmit === 'function') {
```

Comment:
Is the typeof function check necessary? The props already define onSubmit as a function.

Reply:
The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.